docs/learning/ImplementADynamicSchema.md

```diff
@@ -0,0 +1,168 @@
+# Implement A Dynamic Schema
```

Comment:
Jason, did you consider extracting the code from complied examples so our build validates that the code snippets work.  Search the source for `__PUBLISH_EXTRACT_START__` to find some examples.

Also you can use `ts` as the code type for the code block, this is the standard we use in the codebase

Reply:
@ColinKerr I was not aware of either __PUBLISH_EXTRACT_START__ nor ts.  Some of the code blocks I used came from others with no code type at all and I was seeing a markdownlint error for the missing type.  I used 'JavaScript' to make the lint error go away;  I'll replace w/ 'ts' and also look into __PUBLISH_EXTRACT_START__

Reply:
@ColinKerr just replaced ```JavaScript w ```ts  I found some clear examples of PUBLISH_EXTRACT_START and understand what is required.  In only a few places was the code taken as-is from a .ts source file.  Is this pattern a must have?

Reply:
It is important to do in the long run but it can be done as an independent PR